### PR TITLE
[READY] CI now bans files with the same name

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Adds a CI check to ensure we dont have files with the same name in multiple places in the repo

Also disables the DMAPI updater action because it has been broke for ages

## Why It's Good For The Repo
DM runtimes only have the raw filename in them, not the full path, which is a pain in the ass for things with multiple names
image

On top of this, if we say `Go look at life.dm`, the first response will be `Ok which?`, which isn't always easy to deal with. This PR solves both of those

## Testing
Ran it locally, got an output

## Changelog
Nothing player facing

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Part 1 is done, part 2 will come later
image

Atleast this can test that CI flags up issues when they arise

**Review by warriorstar-orion — COMMENTED** [review]
I know this is still in draft but I assume the remaining file renames are the "draft" part

File: tools/ci/check_file_names.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#!/usr/bin/env python3
+
+import glob
+
+all_dm_files = glob.glob(r"**/*.dm", recursive=True)
+
+file_name_map = {}
+
+def main():
+    for file in all_dm_files:
```
[the remark this answers is not in the document]
Comment (AffectedArc07) [reply tools/ci/check_file_names.py:11]: Implemented

File: tools/ci/check_file_names.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        newname = file.replace("\\", "/") # OS normalisation
+        end_name = newname.split("/")[-1] # Get last split
+
+        if end_name not in file_name_map:
+            file_name_map[end_name] = []
+
+        file_name_map[end_name].append(newname) # Take the full thing
+
+    alert_sent = False
+    for filename in file_name_map:
```
[the remark this answers is not in the document]
Comment (AffectedArc07) [reply tools/ci/check_file_names.py]: Implemented

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -5,6 +5,7 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 	name = "item"
 	icon = 'icons/obj/items.dmi'
 	blocks_emissive = EMISSIVE_BLOCK_GENERIC
+	mouse_drag_pointer = MOUSE_ACTIVE_POINTER
```
Comment (S34NW) [inline code/game/objects/items.dm:8]: The fact that this was in some random file by itself is horrific

**Comment by warriorstar-orion** [comment]
> Looks fine to me and my limited python knowledge, maybe worth having @warriorstar-orion look over it a second time before full approval?

Python LGTM.

**Review by S34NW — APPROVED** [review]
deconflict and this can go

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**